libcxx/include/__config

```diff
@@ -14,6 +14,7 @@
 #include <__configuration/abi.h>
 #include <__configuration/availability.h>
 #include <__configuration/compiler.h>
+#include <__configuration/cxx03.h>
```

Comment:
Could we move `_LIBCPP_CXX03_LANG` into `language.h` instead? And it might be worth making a separate patch since that's so easy to do.

So in many places the includes would become `#include <__configuration/language.h>` at the beginning of files.